test/root/scripts/test_function_coercion.ash

```diff
@@ -0,0 +1,38 @@
+/* Uses string format_date_time(string, string, string) as reference */
+string str_in_format = "YYYY";
+string str_date = "2000";
+string str_out_format = "YY";
+
+print("> original format_date_time(string-string-string)");
+print(format_date_time(str_in_format, str_date, str_out_format)); /*Should print "00"*/
+print();
+
+/* Define 2 different layers of typedef */
+typedef string str2;
+typedef str2 str3;
+
+str2 str2_in_format = str_in_format;
+str2 str2_date = str_date;
+str2 str2_out_format = str_out_format;
+str3 str3_in_format = str2_in_format;
+str3 str3_date = str2_date;
+str3 str3_out_format = str2_out_format;
+
+print("> pre-overload; all coerce to original");
+print(format_date_time(str2_in_format, str2_date, str2_out_format));
+print(format_date_time(str3_in_format, str3_date, str3_out_format));
+print(format_date_time(str_in_format, str2_date, str3_out_format));
+print();
+
+string format_date_time(str2 inFormat, str2 dateTimeString, str2 outFormat) {
```

Comment:
Are we sure we want to continue supporting this behavior, but only in this nonintuitive way?

Either we should want to allow script authors to override builtin functions, or we should disallow it entirely. But we shouldn't only allow it via a typedef backdoor.

Reply:
Disallowing it has the potential to break some scripts...

I'm thinking of the inconvenience it would put on people who learn that their function now needs a new name: we don't have a "rename this method" service, so they'd need to manually rename them all...

At the same time, it's not like we're expecting many people to have done that...